src/Shared/ReuseableStringBuilder.cs

```diff
@@ -75,11 +81,22 @@ public int Length
         /// </summary>
         string OpportunisticIntern.IInternable.ExpensiveConvertToString()
         {
+            if( _cachedString == null)
             {
-                return ((ReuseableStringBuilder)this).ToString();
+                _cachedString = ((ReuseableStringBuilder)this).ToString();
```

Comment:
Wouldn't it be better to cache directly in the ToString method? I didn't check, but what if there are ToString calls before Interning happens.